src/lib.rs

```diff
@@ -1,26 +1,21 @@
-#![feature(external_doc)]
-#![doc(include = "../README.md")]
-#![deny(missing_docs)]
-#![cfg_attr(test, deny(warnings))]
-#![cfg_attr(test, feature(plugin))]
-#![cfg_attr(test, plugin(clippy))]
-
-extern crate console;
-extern crate failure;
-extern crate unindent;
-
-use console::style;
-use failure::Error;
+#![cfg_attr(feature = "nightly", deny(missing_docs))]
+#![cfg_attr(feature = "nightly", feature(external_doc))]
+#![cfg_attr(feature = "nightly", doc(include = "../README.md"))]
+
+extern crate termcolor;
+
+use std::error::Error;
 use std::panic;
-use unindent::unindent;
 
 /// Catch any error handlers that occur, and
 // Cargo env vars available:
 // https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates
-pub fn catch_unwind<F: FnOnce() -> Result<(), Error>>(f: F) {
+pub fn catch_unwind<F: FnOnce() -> Result<(), Box<Error>>>(f: F) {
```

Comment:
Curious: why did you opt to replace `failure::Error` with `Box<Error>`? The former should guarantee we always get a stack trace, which I figured would be useful for the reporting.

Reply:
Oh, is this why you used failure? I'm not sure that's what's happening: Failure will only give you stack traces for errors that occur in the `catch_unwind` code -- everything else is part of the `panic_info`, right?

Reply:
yeah, that's totally true! - the way I'm envisioning this though is that this would run in the outer most scope of `fn main()` to catch everything that happens inside of it. And then like with `quicli` it also handles errors that have been returned. Does that make sense?

Reply:
Hmmm. I see what you mean, but I don't think this is how it works. Failure's backtrace will only collect traces when an error is actually propagated -- but when you `panic!`, that doesn't happen (instead, Rust will unwind the stack; unless you set `panic = "abort").

I'm not an authority on how to _really_ solve this, though!